packagedb/tests/test_purlcli.py

```diff
@@ -0,0 +1,524 @@
+import json
+import os
+import sys
+
+import pytest
+from commoncode.testcase import FileBasedTesting
+from django.test import TestCase
+from rest_framework import status
+from rest_framework.test import APIClient
+
+import purlcli
+from packagedb.models import Package
+
+
+class TestPURLCLI_validate(object):
+    @pytest.mark.parametrize(
```

Comment:
@pombredanne How do you feel about using the `pytest.mark.parametrize` to hold the test input and expected results?

Reply:
This works. If the list is small you can use it like there:
- https://github.com/nexB/scancode-toolkit/blob/f70bbb7d9d9bab40a9d504e664bc945b6a1630e8/tests/textcode/test_markup.py#L41
- if this becomes more involved, use a small class to keep the test data expressive and readable.... long lists of tuples are NOT readable: https://github.com/nexB/python-inspector/blob/0ad23abfe051a13afb3126c4112a54ea0288e118/tests/test_utils_pypi.py#L21
- if you have a lot of test/excepted file pairs, this can work https://github.com/nexB/pip-requirements-parser/blob/6629b466001c53ff0def85e42100ca3510023314/tests/test_requirements_parser.py

(For very large test suites, the toolkit approach using YAML test data files for license and copyright can work too, but is much more work and I would likely rewrite this using  pytest.mark.parametrize today though the bulk would stay the same: data files for expectations, test files for data and objects to read these )